src/operators/flat_map.rs

```diff
@@ -34,9 +34,12 @@ where
         input_changes
             .iter()
             .flat_map(|change| {
-                (self.f)(change.item().clone())
-                    .into_iter()
-                    .map(move |item| change.as_ref().map(|_| item))
+                (self.f)(change.item.clone()).into_iter().map(move |item| {
```

Comment:
I think this can be implemented without clone.

Reply:
The item should be moved into the function. This will require making a copy of the count